datafusion/functions-aggregate-common/src/tdigest.rs

```diff
@@ -641,10 +641,19 @@ impl TDigest {
             v => panic!("invalid centroids type {v:?}"),
         };
 
-        let max = cast_scalar_f64!(&state[3]);
-        let min = cast_scalar_f64!(&state[4]);
+        let mut max = cast_scalar_f64!(&state[3]);
+        let mut min = cast_scalar_f64!(&state[4]);
 
-        assert!(max.total_cmp(&min).is_ge());
+        if !min.is_finite() {
```

Comment:
Would it be reasonable to filter out NaN values in `merge_sorted_f64` / `merge_unsorted_f64` functions?

Reply:
@findepi Yeah, I think that probably should be the goal here.